### PR TITLE
Fix hot update for pipeline effects and Float-output-process flow

### PR DESCRIPTION
**Pull request by troublemaker52025**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by troublemaker52025** [comment]
@cocos-robot run test cases

File: cocos/rendering/custom/builtin-pipelines.ts [hunk 1]
```diff
@@ -54,6 +54,9 @@ import {
 import { Feature } from '../../gfx';
 
 export class ForwardPipelineBuilder implements PipelineBuilder {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    onGlobalPipelineStateChanged (): void {}
+
```
Comment (star-e) [inline cocos/rendering/custom/builtin-pipelines.ts]: Empty default implementation is not required.

File: cocos/rendering/custom/builtin-pipelines.ts [hunk 2]
```diff
@@ -79,6 +82,9 @@ export class ForwardPipelineBuilder implements PipelineBuilder {
 }
 
 export class DeferredPipelineBuilder implements PipelineBuilder {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    onGlobalPipelineStateChanged (): void {}
+
```
Comment (star-e) [inline cocos/rendering/custom/builtin-pipelines.ts]: Empty default implementation is not required.

File: cocos/rendering/custom/custom-pipeline.ts [hunk 3]
```diff
@@ -42,6 +42,9 @@ import { CSMLayers } from '../shadow/csm-layers';
 const copyPair = new CopyPair();
 const pairs = [copyPair];
 export class CustomPipelineBuilder implements PipelineBuilder {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    onGlobalPipelineStateChanged (): void {}
+
```
Comment (star-e) [inline cocos/rendering/custom/custom-pipeline.ts]: Empty default implementation is not required.

File: cocos/rendering/custom/custom-pipeline.ts [hunk 4]
```diff
@@ -127,6 +130,10 @@ export class TestPipelineBuilder implements PipelineBuilder {
     constructor (pipelineSceneData: PipelineSceneData) {
         this._sceneInfo = new SceneInfo(pipelineSceneData);
     }
+
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    onGlobalPipelineStateChanged (): void {}
+
```
Comment (star-e) [inline cocos/rendering/custom/custom-pipeline.ts]: Empty default implementation is not required.

File: cocos/rendering/post-process/passes/base-pass.ts [hunk 5]
```diff
@@ -111,6 +111,9 @@ export abstract class BasePass {
         return this.enable;
     }
 
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    onGlobalPipelineStateChanged (): void {}
+
```
Comment (star-e) [inline cocos/rendering/post-process/passes/base-pass.ts]: Empty default implementation is not required.

**Review by jk20012001 — APPROVED** [review]

**Review by star-e — APPROVED** [review]

**Pull request merged**